src/main/java/com/github/edouardswiac/zerotier/ZTServiceImpl.java

```diff
@@ -191,6 +188,173 @@ public void deleteNetworkMember(@NotNull String networkId, String address) {
     execute(request);
   }
 
+  //=======================================
+  
+  @Override
+  public ZTStatus status() {
+    final HttpUrl targetUrl = baseApiUrl.newBuilder().addPathSegment("status").build();
+    final Request request = new Request.Builder().url(targetUrl).build();
+    return get(request, ZTStatus.class);
+  }
+
+  @Override
+  public List<ZTNet> getNets() {
+    final HttpUrl targetUrl = baseApiUrl.newBuilder().addPathSegment("network").build();
+    final Request request = new Request.Builder().url(targetUrl).build();
+    return Arrays.asList(get(request, ZTNet[].class));
+  }
+
+  @Override
+  public ZTController controller() {
+    final HttpUrl targetUrl = baseApiUrl.newBuilder().addPathSegment("controller").build();
+    final Request request = new Request.Builder().url(targetUrl).build();
+    return get(request, ZTController.class);
+  }
+
+  @Override
+  public List<String> getCNetworks() {
+    final HttpUrl targetUrl = baseApiUrl.newBuilder()
+    		.addPathSegment("controller")
+    		.addPathSegment("network").build();
+    final Request request = new Request.Builder().url(targetUrl).build();
+    return Arrays.asList(get(request, String[].class));
+  }
+
+  @Override
+  public ZTCNetwork getCNetwork(@NotNull String networkId) {
+    final HttpUrl targetUrl = baseApiUrl.newBuilder()
+    		.addPathSegment("controller")
+    		.addPathSegment("network")
+    		.addPathSegment(networkId)
+    		.build();
+    final Request request = new Request.Builder().url(targetUrl).build();
+    return get(request, ZTCNetwork.class);
+  }
+
+  @Override
+  public ZTCNetwork updateCNetwork(@NotNull ZTCNetwork network) {
+    if (network.getNwid() == null)
+      throw new IllegalArgumentException("network ID must not be null");
+
+    final HttpUrl targetUrl = baseApiUrl.newBuilder()
+    		.addPathSegment("controller")
+    		.addPathSegment("network")
+    		.addPathSegment(network.getNwid())
+    		.build();
+
+    final Request request = new Request.Builder()
+    		.url(targetUrl)
+    		.post(RequestBody.create(MEDIA_TYPE_JSON, gson.toJson(network)))
+    		.build();
+    return get(request, ZTCNetwork.class);
+  }
+
+  @Override
+  public void deleteCNetwork(@NotNull String networkId) {
+  	final HttpUrl targetUrl = baseApiUrl.newBuilder()
+  			.addPathSegment("controller")
+  			.addPathSegment("network")
+  			.addPathSegment(networkId)
+  			.build();
+
+  	final Request request = new Request.Builder()
+  			.url(targetUrl)
+  			.delete()
+  			.build();
+  	try {
+  		execute(request);
+  	} catch (ZTServiceException ex) {
+  		// catch the 500 that's returned when deleting :/
```

Comment:
You decide.
I am not really familiar with cooperating on github best practices (this is the first time I issue a pull-request), so I'm not in position to take sensible decisions on the subject.
Please tell me what is the best way to cooperate.